homeassistant/components/togrill/coordinator.py

```diff
@@ -74,13 +74,22 @@ def __init__(
             name="ToGrill",
             update_interval=SCAN_INTERVAL,
         )
-        self.address = config_entry.data[CONF_ADDRESS]
+        self.address: str = config_entry.data[CONF_ADDRESS]
         self.data = {}
         self.device_info = DeviceInfo(
```

Comment:
I think it's weird that the coordinator has a device info and another device info and that the entity picks the one to use. It's unclear from reading the coordinator what is used when.

Reply:
Did it get clearer with the change to a function?